eodatasets3/wagl.py

```diff
@@ -121,6 +135,161 @@ def _unpack_products(
                     )
 
 
+def missing_packets_xml(granule):
+    """
+    Return children xml document if missing packets and None if not
+    """
+    l1_zip_path = Path(granule.wagl_metadata["source_datasets"]["source_level1"])
+    xml_file_path = Path(
+        granule.source_level1_metadata.accessories["metadata:s2_tile"].path
```

Comment:
This is great, I forgot that the metadata gave us the tile path. So it should handle all the various package layouts of S2 automatically, including multi-granule.